src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -750,5 +733,328 @@ private static string ToHtmlCase(string name)
+            }
+
+            private TagHelperRequiredAttributeDescriptor ParsePlainSelector(ErrorSink errorSink)
+            {
+                var nextSeparator = _requiredAttributes.IndexOf(',', _index);
+                string attributeName;
+
+                if (nextSeparator == -1)
+                {
+                    attributeName = _requiredAttributes.Substring(_index).Trim();
+                    _index = _requiredAttributes.Length;
+                }
+                else
+                {
+                    attributeName = _requiredAttributes.Substring(_index, nextSeparator - _index).Trim();
+                    _index = nextSeparator;
+                }
+
+                var hasWildcard = false;
+                if (attributeName[attributeName.Length - 1] == RequiredAttributeWildcardSuffix)
+                {
+                    attributeName = attributeName.Substring(0, attributeName.Length - 1);
+                    hasWildcard = true;
+                }
+
+                TagHelperRequiredAttributeDescriptor descriptor = null;
+                if (ValidateName(attributeName, targetingAttributes: true, errorSink: errorSink))
+                {
+                    descriptor = new TagHelperRequiredAttributeDescriptor
+                    {
+                        Name = attributeName
+                    };
+
+                    if (hasWildcard)
+                    {
+                        descriptor.Operator = RequiredAttributeWildcardSuffix;
+                    }
+                }
+
+                return descriptor;
+            }
+
+            private string ParseCSSAttributeName(ErrorSink errorSink)
+            {
+                var nameStartIndex = _index;
+                while (!AtEnd && !char.IsWhiteSpace(Current) && !At('=') && !At(']'))
+                {
+                    Next();
+                }
+                var nameEndIndex = _index;
+
+                PassOptionalWhitespace();
+
+                var attributeNameLength = 0;
+                if (At('='))
+                {
+                    attributeNameLength = nameEndIndex - nameStartIndex;
+
+                    // If this proves true it means there was 0 whitespace between the attribute name and the '='.
+                    // Therefore, the operator would be baked into the attribute name by default.
+                    var potentialOperator = _requiredAttributes[_index - 1];
+                    if (TagHelperRequiredAttributeDescriptor.IsSupportedCSSValueOperator(potentialOperator))
+                    {
+                        attributeNameLength--;
+                    }
+                }
+                else if (At(']'))
+                {
+                    attributeNameLength = nameEndIndex - nameStartIndex;
+                }
+                else if (NextIs('=') && TagHelperRequiredAttributeDescriptor.IsSupportedCSSValueOperator(Current))
+                {
+                    // Move past selector
```

Comment:
Don't move past the operator or equal sign. Leave those characters for `ParseCSSValueOperator()`. In other word, just stop when at an `'='` or supported operator. That way `ParseCSSValueOperator()` can use its `errorSink` and confirm `'='` follows a supported character. (This also avoids backing up while parsing.)